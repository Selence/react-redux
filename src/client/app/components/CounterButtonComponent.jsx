import React from 'react';
import PropTypes from 'prop-types';

/**
 * @param {Object} actions
 * @returns {XML}
 */
export const CounterButtonComponent = ({actions}) => {

    const { increaseCounter } = actions;

    return (
        <div
            className="counter__button"
            onClick={increaseCounter}>
            Increase Counter +1
        </div>
    )

};

CounterButtonComponent.propTypes = {
    actions: PropTypes.shape({
        increaseCounter: PropTypes.func
    })
};

