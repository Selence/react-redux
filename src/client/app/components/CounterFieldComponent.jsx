import React from 'react';
import PropTypes from 'prop-types';

/**
 * @param {Number} counter
 * @returns {XML}
 */
export const CounterFieldComponent = ({
    counter
}) => {

    const renderHint = () => (
        <div className="hint">
            <p className="hint__text">Download <a href="https://github.com/zalmoxisus/redux-devtools-extension">redux-devtools-extension</a> to show the store</p>
        </div>
    );

    return (
        <div>
            <p>Counter:</p>
            <div className="counter__field">{counter}</div>
            {counter > 1 && renderHint()}
        </div>
    );
};

CounterFieldComponent.propTypes = {
    counter: PropTypes.number
};
