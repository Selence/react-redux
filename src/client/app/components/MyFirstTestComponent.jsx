import React from 'react';
import PropTypes from 'prop-types';

/**
 * @returns {XML}
 */
export const MyFirstTestComponent = ({
    counter
}) => {
    return (
        <div>
            uh...Yes I can!
        </div>
    );
};

MyFirstTestComponent.propTypes = {
    counter: PropTypes.number
};
