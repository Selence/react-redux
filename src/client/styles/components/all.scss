@import "counter";
@import "app";
@import "hint";