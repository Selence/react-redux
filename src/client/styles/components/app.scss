.app {
  width: 100%;

  @include element('text') {
    font-size: 25px;
  }
}