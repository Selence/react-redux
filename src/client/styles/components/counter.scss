.counter {
  width: 20%;

  @include element('button') {
    @include nonSelectable();

    background-color: lightgray;
    border: 1px solid lightgray;
    padding: 20px;
    text-align: center;
    width: 100%;

    &:hover {
      border-color: black;
    }

    &:active {
      background-color: white;
    }
  }

  @include element('field') {
    border: 1px dashed black;
    width: 10%;
  }
}