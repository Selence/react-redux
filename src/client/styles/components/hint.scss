.hint {
  border: 1px solid gray;
  margin-top: 10px;

  @include element('text') {
    padding: 15px;
  }
}