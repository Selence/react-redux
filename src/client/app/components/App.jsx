import React, { Component } from 'react';
import { MyFirstTestComponent } from './MyFirstTestComponent';

/**
 * @returns {XML}
 */
export class App extends Component {
    constructor(props) {
        super(props);
    }

    /**
     * @returns {XML}
     */
    render () {
        return (
            <div>
                <p>Hello React Project ... can you load ... ?</p>
                <MyFirstTestComponent />
            </div>
        );
    }
}