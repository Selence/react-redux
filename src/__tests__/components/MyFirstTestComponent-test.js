import React from 'react';
import renderer from 'react-test-renderer';
import { CounterFieldComponent } from '../../client/app/components/CounterFieldComponent'

describe('Component CounterField', () => {
    const renderComponent = () => (
        renderer.create(<CounterFieldComponent />).toJSON()
    );

    describe('when the component is called', () => {
        it('renders correctly', () => {
            expect(renderComponent()).toMatchSnapshot();
        });
    })
});