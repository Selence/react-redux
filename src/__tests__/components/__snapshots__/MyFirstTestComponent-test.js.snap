// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CounterField when the component is called renders correctly 1`] = `
<div>
  <p>
    Counter:
  </p>
  <div
    className="counter__field"
  />
</div>
`;

exports[`MyFirstTestComponent when the component is called renders correctly 1`] = `
<divs>
  uh...Yes I can!
</divs>
`;
